beta-src/src/utils/map/invalidClick.ts

```diff
@@ -0,0 +1,38 @@
+import * as d3 from "d3";
+
+export default function invalidClick(
+  evt: any, // FIXME: what type is this?
+  territoryName: string,
+): void {
+  console.log("INVALID CLICK");
+  const territorySelection = d3.select(`#${territoryName}-territory`);
+  const territory: SVGSVGElement = territorySelection.node();
+  if (territory) {
+    const screenCTM = territory.getScreenCTM();
+    if (screenCTM) {
+      const pt = territory.createSVGPoint();
+      pt.x = evt.clientX;
+      pt.y = evt.clientY;
+      const { x, y } = pt.matrixTransform(screenCTM.inverse());
+      territorySelection
+        .append("circle")
+        .attr("cx", x)
+        .attr("cy", y)
+        .attr("r", 6.5)
+        .attr("fill", "red")
+        .attr("fill-opacity", 0.4)
+        .attr("class", "invalid-click");
+      territorySelection
+        .append("circle")
+        .attr("cx", x)
+        .attr("cy", y)
+        .attr("r", 14)
+        .attr("fill", "red")
+        .attr("fill-opacity", 0.2)
+        .attr("class", "invalid-click");
+      setTimeout(() => {
+        d3.selectAll(".invalid-click").remove();
+      }, 100);
+    }
+  }
```

Comment:
Can this be done the territory component itself. e.g. by passing a prop that says it's invalid.  It's not ideal to be updating the component using DOM.

Reply:
Ok I see that this code was just moved. None the less it would be cleaner to do this in he component that creates the element. 

```d3.select(`#${territoryName}-territory`);```

Reply:
Yeah, all this imperative drawing of things is non-ideal, but I don't think it's at the top of the list for refactoring, especially because the drawing code here seems to work well as is and is complicated enough that it would be substantial work to reimplement.